kernel/kernel-api/src/test/scala/io/delta/kernel/internal/icebergcompat/IcebergCompatV2MetadataValidatorAndUpdaterSuite.scala

```diff
@@ -161,6 +161,50 @@ class IcebergCompatV2MetadataValidatorAndUpdaterSuite
     validateAndUpdateIcebergCompatV2Metadata(isNewTable, metadata, protocol)
   }
 
+  test("compatible type widening is allowed with icebergCompatV2") {
+    val schema = new StructType()
+      .add(
+        new StructField(
+          "intToLong",
+          IntegerType.INTEGER,
+          true,
+          FieldMetadata.empty()).withTypeChanges(
+          Seq(new TypeChange(IntegerType.INTEGER, LongType.LONG)).asJava))
+      .add(
+        new StructField(
+          "decimalToDecimal",
+          new DecimalType(10, 2),
+          true,
+          FieldMetadata.empty()).withTypeChanges(
+          Seq(new TypeChange(new DecimalType(5, 2), new DecimalType(10, 2))).asJava))
+
+    val metadata = getCompatEnabledMetadata(schema)
+    val protocol = getCompatEnabledProtocol(TYPE_WIDENING_RW_FEATURE)
+
+    // This should not throw an exception
+    validateAndUpdateIcebergCompatV2Metadata(false, metadata, protocol)
+  }
+
+  test("incompatible type widening throws exception with icebergCompatV2") {
```

Comment:
assuming `TypeWideningChecker.isIcebergV2Compatible` already has the complete coverage for all different type changes?

Reply:
Yes.  In https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/test/scala/io/delta/kernel/internal/types/TypeWideningCheckerSuite.scala I added some more test cases to make the IcebergV2 tests more robust.